Hide API key.

carlthronson opened the pull request:
When running locally, set these values in .env.local

API_KEY=whatever
API_URL=http://localhost:8082/graphql

When running on Vercel, set these values in the environment

API_KEY=<actual_backend_api_key>
API_URL=<actual_backend_server_url>

carlthronson committed:
API_KEY and API_URL.  Put these in .env.local

API_KEY=whatever
API_URL=http://localhost:8082/graphql

When it runs on Vercel, these values will be set in the environment

lean1190 commented on next.config.mjs:
```diff
@@ -1,4 +1,11 @@
 /** @type {import('next').NextConfig} */
-const nextConfig = {};
+const nextConfig = {
+  env: {
+    RESEND_API_KEY: "re_ZacMr3bG_MZCB3tg788UrUieq3XgMh3hX",
+    API_KEY: process.env.API_KEY,
+    //when the server is run locally, I believe this key is "whatever"
+    API_URL: process.env.API_URL,
+  },
```
Should we use an .env file instead?

carlthronson replied:
I'm not the most experienced with frontend coding.  So you may find a better way to do this.  But what I'm trying to do is (a) remove the api key from github and (b) remove the api key from the browser.

lean1190 commented on next.config.mjs:
```diff
@@ -1,4 +1,11 @@
-const nextConfig = {};
+const nextConfig = {
+  env: {
+    RESEND_API_KEY: "re_ZacMr3bG_MZCB3tg788UrUieq3XgMh3hX",
```
Could we also move this to an env variable?
It's not ideal to have this value exposed

carlthronson replied:
Yeah, sorry, I was just trying to fix the api key.  Sometimes I have a focus that is too narrow.  But you're right we should create a PR for that as well.

lean1190 replied:
I will add a `.env.example` file and remove this from here.
Then, each person needs to create a `.env.local` file on their machines with the right values.

@carlthronson do you know where are we storing these API keys and URLs?

carlthronson replied:
If you are running the system locally, then, you can configure the api key in the backend, and you just need to match it in the frontend.  The api url is of the form http://localhost:8082/graphql.   So you might change the port number.  For the systems running in the cloud, the values are configured in the cloud.  And that is the only place where they can be found.  We do not want these values to be available in github.

lean1190 commented on src/app/api/graphql/route.tsx:
```diff
@@ -0,0 +1,36 @@
+import { NextRequest, NextResponse } from 'next/server';
+
+export async function POST(request: NextRequest) {
```
This is a very cool approach, nice!

carlthronson replied:
As I said, I'm not very experienced with frontend coding.  I probably got this code from doing some research.  But, it seems to work. :)

lean1190 committed:
Remove env variables from config and add example env file

This means each person will have to duplicate this .env.example and rename it to .env.local with appropriate values

lean1190 committed:
site label: fix image path
